src/cascadia/TerminalSettingsEditor/ColorSchemesPageViewModel.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "ColorSchemesPageViewModel.h"
+#include "ColorSchemesPageViewModel.g.cpp"
+
+#include <LibraryResources.h>
+#include "..\WinRTUtils\inc\Utils.h"
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    static const std::array<std::wstring, 9> InBoxSchemes = {
+        L"Campbell",
+        L"Campbell Powershell",
+        L"Vintage",
+        L"One Half Dark",
+        L"One Half Light",
+        L"Solarized Dark",
+        L"Solarized Light",
+        L"Tango Dark",
+        L"Tango Light"
+    };
+
+    ColorSchemesPageViewModel::ColorSchemesPageViewModel(const Model::CascadiaSettings& settings) :
+        _settings{ settings }
+    {
+        _MakeColorSchemeVMsHelper();
+        CurrentScheme(_AllColorSchemes.GetAt(0));
+    }
+
+    void ColorSchemesPageViewModel::RequestSetCurrentScheme(Editor::ColorSchemeViewModel scheme)
+    {
+        CurrentScheme(scheme);
+
+        // Update the state of the page
+        _PropertyChangedHandlers(*this, Windows::UI::Xaml::Data::PropertyChangedEventArgs{ L"CanDeleteCurrentScheme" });
+    }
+
+    bool ColorSchemesPageViewModel::RequestRenameCurrentScheme(winrt::hstring newName)
+    {
+        // check if different name is already in use
+        const auto oldName{ CurrentScheme().Name() };
+        if (newName != oldName && _settings.GlobalSettings().ColorSchemes().HasKey(newName))
+        {
+            return false;
+        }
+        else
+        {
+            // update the settings model
+            CurrentScheme().Name(newName);
+            _settings.GlobalSettings().RemoveColorScheme(oldName);
+            _settings.GlobalSettings().AddColorScheme(CurrentScheme().SettingsModelObject());

```

Comment:
I don't like exposing the scheme Settings Model Object like this, but we need to be able to pass it to `GlobalSettings` to actually add the color scheme... maybe the `ColorSchemesPageViewModel` should instead hold a map of `ColorSchemeViewModels` : `ColorSchemes` so it does not need to ask the `ColorSchemeViewModel` for the settings model object beneath it?